### PR TITLE
*: make sink method FlushRowChangedEvents non-blocking

### PR DESCRIPTION
**Pull request by asddongmen**
### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: cdc/processor/pipeline/sink.go [hunk 1]
```diff
@@ -136,10 +136,17 @@ func (n *sinkNode) flushSink(ctx pipeline.NodeContext, resolvedTs model.Ts) (err
...
 		return errors.Trace(err)
 	}
-	checkpointTs, err := n.sink.FlushRowChangedEvents(ctx, resolvedTs)
-	if err != nil {
-		return errors.Trace(err)
+
+	var checkpointTs uint64
+	var ok bool
+	// retry until ok is true
+	for !ok {
```
Comment (sdojjy) [inline cdc/processor/pipeline/sink.go:143]: why retry here?

Comment (asddongmen) [reply cdc/processor/pipeline/sink.go:143]: If not retry here, the `resolvedTs` pass in this `Flush` may be lost.

File: cdc/processor/pipeline/sink.go [hunk 2]
```diff
@@ -131,22 +134,24 @@ func (n *sinkNode) flushSink(ctx pipeline.NodeContext, resolvedTs model.Ts) (err
 		resolvedTs = n.targetTs
 	}
 	if resolvedTs <= n.checkpointTs {
-		return nil
+		return ok, nil
```
Comment (lonng) [inline cdc/processor/pipeline/sink.go:137]: I think we use `true/false` directly if more readable. I don't think it's make sense to define a boolean `ok` and return `ok/!ok`.

**Comment by liuzix** [comment]
require LGT3 because the PR changes public API.

File: cdc/processor/pipeline/sink.go (hunk 2 above)
Comment (asddongmen) [reply cdc/processor/pipeline/sink.go:137]: Reasonable suggestion, thanks~

**Comment by asddongmen** [comment]
find  better way #3541

**Pull request closed without merging**